x-pack/plugins/secops/public/components/formatted_ip/index.tsx

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { isArray, isEmpty, isString, uniq } from 'lodash/fp';
+import * as React from 'react';
+import { pure } from 'recompose';
+
+import { escapeQueryValue } from '../../lib/keury';
+import { DragEffects, DraggableWrapper } from '../drag_and_drop/draggable_wrapper';
+import { escapeDataProviderId } from '../drag_and_drop/helpers';
+import { getOrEmptyTagFromValue } from '../empty_value';
+import { IPDetailsLink } from '../links';
+import { parseQueryValue } from '../timeline/body/renderers';
+import { DataProvider } from '../timeline/data_providers/data_provider';
+import { Provider } from '../timeline/data_providers/provider';
+import { TruncatableText } from '../truncatable_text';
+
+const getUniqueId = ({
+  contextId,
+  eventId,
+  fieldName,
+  address,
+}: {
+  contextId: string;
+  eventId: string;
+  fieldName: string;
+  address: string | object | null | undefined;
+}) => `id-${contextId}-${fieldName}-${address}-for-event-${eventId}`;
+
+const tryStringify = (value: string | object | null | undefined): string => {
+  try {
+    return JSON.stringify(value);
+  } catch (_) {
+    return `${value}`;
+  }
+};
+
+const getDataProvider = ({
+  contextId,
+  eventId,
+  fieldName,
+  address,
+}: {
+  contextId: string;
+  eventId: string;
+  fieldName: string;
+  address: string | object | null | undefined;
+}): DataProvider => ({
+  enabled: true,
+  id: escapeDataProviderId(getUniqueId({ contextId, eventId, fieldName, address })),
+  name: `${fieldName}: ${parseQueryValue(address)}`,
+  queryMatch: {
+    field: fieldName,
+    value: escapeQueryValue(parseQueryValue(address)),
+  },
+  excluded: false,
+  kqlQuery: '',
+  and: [],
+});
+
+const NonDecoratedIp = pure<{
+  contextId: string;
+  eventId: string;
+  fieldName: string;
+  value: string | object | null | undefined;
+  width?: string;
+}>(({ contextId, eventId, fieldName, value, width }) => (
+  <DraggableWrapper
+    key={getUniqueId({ contextId, eventId, fieldName, address: value })}
+    dataProvider={getDataProvider({ contextId, eventId, fieldName, address: value })}
+    render={(dataProvider, _, snapshot) =>
+      snapshot.isDragging ? (
+        <DragEffects>
+          <Provider dataProvider={dataProvider} />
+        </DragEffects>
+      ) : typeof value !== 'object' ? (
```

Comment:
I do not think is not going to do what you think it is. You are going to get strings of `null` on your UI potentially. I really think this has some bugs that unit tests would flush out.

You should fix this with lodash's `isObject(value)` and let me break down why.

Within lodash code for `isObject(value)` it has this bit of magic to it:

https://github.com/lodash/lodash/blob/4.17.11/lodash.js#L11744

```ts
var type = typeof value;
return value != null && (type == 'object' || type == 'function');
```

Which is great because the problem is that
```ts
typeof null == 'object'
true
```
will return true and then JSON.stringify(null) will return "null" and ... getOrEmptyTagFromValue("null") will show a null value.

Let me know if that's not correct.